### ISSUE TITLE
Ractive debug option not propagated to viewmodel

### ISSUE THREAD

**Issue by timoxley** [ractivejs/ractive#1792]
As far as I can tell from the [debug docs](http://docs.ractivejs.org/latest/options#debug), this should report an error, yet it just silently fails (which is basically worst case scenario).

```js
import Ractive from 'ractive'

new Ractive({
  el: document.body,
  template: '{{getError()}}',
  debug: true,
  data: {
    getError: function() {
      throw new Error('what')
    }
  }
})

```

![image](https://cloud.githubusercontent.com/assets/43438/6586477/d1a44320-c7b4-11e4-9954-0339c99a7c93.png)

I can only get it work with `Ractive.defaults.debug = true`.

Similarly to #1788, it'd be really nice if there was a way to configure Ractive to just step out of the way and just let the error happen.

**Comment by Rich-Harris** [comment]
This will be fixed by #1813 - rather than a per-instance/class `debug` option, there'll be a single global `Ractive.DEBUG` flag which defaults to `true`. At the moment, the behaviour will be to warn (with a stack trace) in debug mode, and fail silently in production. The warning looks like this:

![screen shot 2015-03-13 at 11 01 36 pm](https://cloud.githubusercontent.com/assets/1162160/6650061/006eb42a-c9d5-11e4-99d7-244e85e608f2.png)

It could be made to throw an error instead (or throw an error in debug mode, or something like that) - the reason it *doesn't* currently do that is because `{{missingVar}}` is fine in anything-goes Mustache (it just doesn't print anything), so Ractive expressions follow the same pattern. It's convenient if you're (for example) accessing properties of objects that don't exist yet, which is why I'm inclined to think it shouldn't be an error (especially if you get the stack trace in debug mode anyway, albeit without sourcemaps since that only works reliably with errors apparently) but I'm open to persuasion!

Either way, debug-by-default ought to make life easier.

**Comment by Rich-Harris** [comment]
Closing this, as `Ractive.DEBUG` is now implemented (#1813)

**Comment by timoxley** [comment]
Thanks a lot! Very good call on debug by default. :+1:
